packages/vite/src/plugins/vite-plugin-remove-from-bundle.ts

```diff
@@ -0,0 +1,45 @@
+import type { PluginOption } from 'vite'
+
+type ModulesToExclude = Array<{ id: RegExp }>
+
+/**
+ *
+ * This is a vite plugin to remove modules from the bundle.
+ *
+ * Only applies on build, not on dev.
+ *
+ */
+export default function removeFromBundle(
+  modulesToExclude: ModulesToExclude
+): PluginOption {
+  const isMissingIdToExclude = modulesToExclude.some(
+    (module) => module.id === undefined
+  )
+
+  if (isMissingIdToExclude) {
+    throw new Error('You must specify an id to exclude')
+  }
+
+  return {
+    name: 'remove-from-bundle',
+    apply: 'build', // <-- @MARK important
+    load: (id) => {
+      excludeOnMatch(modulesToExclude, id)
```

Comment:
Do we need to `return` here?

Reply:
Yep yep! Must've broken it trying get it working for the test!